src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3067,18 +3025,12 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
             process_variable_init_val(current_scope->get_symbol(var_name), x.base.base.loc, init_expr);
         }
 
-        if (is_allocatable && x.m_value && AST::is_a<AST::Call_t>(*x.m_value)) {
-            tmp = check_to_allocate_array(x.m_value, var_name, x.base.base.loc);
```

Comment:
We have to test these changes in LFortran as well.

Reply:
>We have to test these changes in LFortran as well.

I think we need not test this particular change. I think only the backend changes need to be tested. I submitted https://github.com/lfortran/lfortran/pull/2262 which includes the LibASR changes.